apache/superset
# feat(explore): Allow using time formatter on temporal columns in data table

### SUMMARY
Enable switching between displaying epoch (original value) and time formatted value (format: YYYY-mm-dd HH:MM:SS).

Temporal columns in data table (tabs "Data" and "Samples") now have a clickable "gear" icon, which opens a popover that allows switching between 2 display modes. 

Selected display mode is specific to each column - user can select different display modes for each temporal column. The selection is also specific to datasource, meaning if there are temporal columns that have the same name in 2 datasources, those columns can have different display modes.

Selection is saved in local storage - when user reloads the page, changes hart config etc., it will be persisted.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

https://user-images.githubusercontent.com/15073128/152353312-3abe36c5-1f23-4e72-8afc-991471b7722f.mov

### TESTING INSTRUCTIONS
1. Create a chart with some temporal columns
2. Open data table below the chart - temporal columns should have "gear" icons on their left sides
3. Click the icon and select "Formatted time"
4. Verify that the displayed values for that column have changed from long numbers to human readable dates
5. Reload the page, verify that your selections have been saved
6. Switch between datasources, verify that the selections are specific for each datasource
7. Perform the above testing instructions for charts using V1 API and legacy API

### ADDITIONAL INFORMATION



https://app.shortcut.com/preset/story/33355/explore-south-table-timestamp-column-in-data-panel-did-not-show-correctly

CC @rusackas @jinghua-qa @kasiazjc
